packages/layer1-usdc/src/l-1-usdc-message-transmitter.ts

```diff
@@ -0,0 +1,68 @@
+import { BigInt, Bytes, crypto, ethereum } from "@graphprotocol/graph-ts";
+import {
+  MessageReceived as MessageReceivedEvent,
+  MessageSent as MessageSentEvent,
+} from "../generated/L1USDCMessageTransmitter/L1USDCMessageTransmitter";
+import { MessageReceived, MessageSent } from "../generated/schema";
+import { log } from "matchstick-as";
+
+function leftPadBytes(data: Bytes, length: number): Bytes {
+  const completeData = new Bytes(length as i32);
+  const zeroBytesToFillPrefix = completeData.length - data.length;
+  for (let i = 0; i < completeData.length; i++) {
+    if (i < zeroBytesToFillPrefix) {
+      completeData[i] = 0;
+    } else {
+      completeData[i] = data[i - zeroBytesToFillPrefix];
+    }
+  }
+  return completeData
+}
+
+export function handleMessageReceived(event: MessageReceivedEvent): void {
+  let entity = new MessageReceived(
+    event.transaction.hash.concatI32(event.logIndex.toI32()),
+  );
+  entity.caller = event.params.caller;
+  entity.sourceDomain = event.params.sourceDomain;
+  entity.nonce = event.params.nonce;
+  entity.sender = event.params.sender;
+  entity.messageBody = event.params.messageBody;
+
+  entity.blockNumber = event.block.number;
+  entity.blockTimestamp = event.block.timestamp;
+  entity.transactionHash = event.transaction.hash;
+
+  // Only index messages from Arbitrum
+  if (event.params.sourceDomain.equals(BigInt.fromI32(3))) {
```

Comment:
To save a bit of processing you can make this check the first thing and return if not Arbitrum